test/services/audit/AuditServiceSpec.scala

```diff
@@ -139,6 +148,21 @@ trait AuditTestSupport
       .deepMerge(Json.toJson(allRecordsXmlMarshallingTest).as[JsObject])
   )
 
+  val extendedCancellationEvent = ExtendedDataEvent(
+    auditSource = appConfig.appName,
+    auditType = AuditTypes.Cancellation.toString,
+    tags = AuditExtensions
+      .auditHeaderCarrier(hc)
+      .toAuditTags(
+        transactionName = s"Export-Declaration-${AuditTypes.Cancellation.toString}-payload-request",
+        path = s"customs-declare-exports/${AuditTypes.Cancellation.toString}/full-payload"
+      ),
+    detail = Json
+      .toJson(AuditExtensions.auditHeaderCarrier(hc).toAuditDetails())
+      .as[JsObject]
+      .deepMerge(Json.toJson(cancellationDeclarationTest).as[JsObject])
```

Comment:
This detail looks really complicated inside, can we prepare this in few variable with descripotion what's this?

val a = Json
      .toJson(AuditExtensions.auditHeaderCarrier(hc).toAuditDetails())
      .as[JsObject]

val b= Json.toJson(cancellationDeclarationTest).as[JsObject]

val c = a.deepMerge(b)

To understand what is inside a, b, c

Reply:
This has been addressed.